### DISCUSSION TITLE
Thanks for build 248 supporting Ventura + Hint regarding drag-n-drop linking of images and documents

### DISCUSSION THREAD

**Discussion by ZimTom** [zim-desktop-wiki/zim-desktop-wiki#2325] [Q&A]
Hello @jaap-karssenberg and @dehesselle ,

kudos to both of you for getting build 248 out. I am so glad that the lottery game in terms of mouse behaviour in ZIM is over now. Everything seems to be working perfect - apart from one thing where I don't know whether it is a general problem or just an issue of the MacOS version.

Under WIndows I was used to be able to insert images from the attachment directory of a Wiki page directly by drag-n-drop and the same for links to external documents. Both isn't working as it used to be and I assume these issues are interlinked. Here is what happens.

If I try to insert an image by drag-n-drop, the picture doesn't appear but instead of it a link to the file is inserted in the page.

![Insert_Link_via_ Drag-n-Drop_02_230219-1](https://user-images.githubusercontent.com/108861992/219941719-298c3937-a5d2-40ce-894f-26ec39483609.png)

If I then click on the link I get the message "no such file or folder ..."

The reason for this behaviour are two spaces (or non-displayable characters) that are inserted in the created link behind the file extension.
After deleting these two characters the link works normally and the image is opened in preview.

This happens independent from the file type, so that all the document links created by drag-n-drop need this correction to work.

I would assume that if this is corrected, also inserting images by drag-n-drop will work again automatically (I guess due to adding these two extra characters the file format isn't recognzied as an image format and treats the image as an ordinary document)

Hope this makes sense to you ...

Best regards

Thomas

**Comment by dehesselle** [comment]
I cannot reproduce this issue. Do you have whitespace in your home directory (like `/Users/John Doe`) ?

**Reply by introt** [reply]
One possible avenue for debugging is to log the path with eg. base64 encoding to figure out what the trailing chars are. Does your pipeline allow for easy extra debug builds?

**Comment by ZimTom** [comment]
Sorry, but I do not know what you mean by "whitespace". What I can say is the following:

I don't have the Windows system with ZIM running available anymore, but as far as I can recall from the time switching over from Windows to Mac, dragging and dropping images from the attachment directory, display within the ZIM window always worked normal, i.e. the image appeared om the Wiki page.

In MacOS ist looks like shown in the example. Just in case my orginal description was misleading due to my language, here are two updated examples

<img width="934" alt="ZIM-Example_1_230609-1" src="https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/108861992/9065f1bf-6a2d-4ff5-961a-45d14d74613b">
<img width="1025" alt="ZIM-Example_2_230609-1" src="https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/108861992/24be9bd5-ed95-46a8-ad96-12ebeacc713d">

To me it looks like drag'n drop from the attachment directory display in th ZIM window adds an additional character to the filename, while drag'n drop from the opened finder window doesn't.
If I open the link properties in scenario 1 and copy the entire content of the link field in an empty file in Text Edit it shows the filename with the cursor ending up in the lext line as if a CR was included in the content of the link field.

If I drag'n drop from the finder window, then open the image properties and copy the link field content in an emtpy file in Text Edit it shows the file name of the image WITHOUT moving the cursor to the next line, i.e. the field content in ZIM seems to be the pure filename of the image without any extra character attached to it.

Hope this helps a little ...

It is not a big thing as it doesn't prevent me from inserting images, I just thought i might be worth mentioning as to me it looks to an unusual behaviour compared to the Windows version. That said, I will try to find a way to try it on a Windows system a.s.a.p. just to verify my memories.
